contracts/src/v0.8/KeeperRegistry.sol

```diff
@@ -875,12 +911,30 @@ contract KeeperRegistry is
     address from,
     uint256 id,
     bytes memory performData,
-    bool useTxGasPrice
+    bool isExecution
   ) private view returns (PerformParams memory) {
     uint256 gasLimit = s_upkeep[id].executeGas;
     (uint256 gasWei, uint256 linkEth) = _getFeedData();
-    uint256 adjustedGasWei = _adjustGasPrice(gasWei, useTxGasPrice);
-    uint96 maxLinkPayment = _calculatePaymentAmount(gasLimit, adjustedGasWei, linkEth);
+    uint256 adjustedGasWei = _adjustGasPrice(gasWei, isExecution);
+
+    uint256 l1CostWei = 0;
+    bytes memory data = new bytes(0);
+    if (PAYMENT_MODEL == PaymentModel.OPTIMISM) {
+      if (isExecution) {
+        data = bytes.concat(msg.data, L1_FEE_DATA_PADDING);
+      } else {
+        data = bytes.concat(ESTIMATED_MSG_DATA, L1_FEE_DATA_PADDING);
```

Comment:
this code path can be invoked by checkUpkeep when isExecution is false. In this case, we know the performData. another approach will be concat performData, L1_FEE_DATA_PADDING, and some more paddings which is used to offset the difference between performData and msg.data from performUpkeep. i feel that will be more accurate overall but more complex.

Reply:
I think perform data would be more accurate here, even though it will have 4 less bytes than actual msg.data

Reply:
Done